### ISSUE TITLE
New feature: Inline links - simpler syntax support

### ISSUE THREAD

**Issue by diethardsteiner** [MacDownApp/macdown#65]
I think the way inline links are currently implemented is way to difficult to write. I don't really want to use HTML for this purpose if there is another way. Maybe you can offer support for existing Markdown extensions that cover that functionality. 
I am mainly familiar with MultiMarkdown and if you define in example a heading there like this:

```
# My heading
```

You can just link to it using this:

```
[My heading]()
```

I think that's way easier.

**Comment by RoyiAvital** [comment]
Would be a real great addition.
Hopefully will be implemented in howdown.

**Comment by Jmuccigr** [comment]
Pandoc has this feature. Some consistency across the markdown world would be nice. :-)

**Comment by RoyiAvital** [comment]
@Jmuccigr , I think this is why they invented [CommonMark](http://commonmark.org/).

I guess we all would like MacDown to support CommonMark.

**Comment by Jmuccigr** [comment]
@RoyiAvital,

Yes, the developer of pandoc is on the common mark team as well. CommonMark isn't finalized yet either, so it's not so simple.

**Comment by RoyiAvital** [comment]
I see...
Hopefully they also add features beyond the ones [they listed](http://commonmark.org/help/).

I wish they added:

1. Math.
2. Tables.
3. To Do List (Like in GitHub).
4. Tab for Text Indention (Aligning text like we do for code).
5. Alignment of Images.
6. Commenting mechanism.
7. Footnotes.
8. Reference system.
